### ISSUE TITLE
Simple TIP Bot

### ISSUE THREAD

**Issue by shawntabrizi** [ChaosDAO-org/InternalGov-2.0#77]
A simple TIP bot could be included into the project, allowing DAOs to discuss community members who have done things for Polkadot, and should receive a tip for that work.

Basic idea:

- there should be a discord channel called "tips"
- any user can create a new thread in that channel
    - perhaps with some limit, like only once per 24 hours, but people in the DAO shouldnt spam anyway, or they would be kicked out.
- the beginning of the thread should be some evidence or text of the work the person has done, and include the tip recipient's address
- other members of the dao submit a post with a "tip" amount.
    - for example `tip: 10 DOT` or `tip: 0 DOT`
    - perhaps we should allow a justification string aswell
- The bot then waits 48 hours or for everyone to vote, and uses the **median** value to determine how much to tip.
    - Perhaps the bot shows the mean, median, mode
- The bot then opens a tip for that amount using the DAO account, and votes AYE on the tip.
    - perhaps all "justification" strings are included in the tip anonymously to give context on why the DAO has decided to tip
    
    
Something like this. Would be happy to contribute if I was given some starting code on entry point to write the logic.

**Comment by shawntabrizi** [comment]
Basically a more "voting" based version of: https://github.com/paritytech/substrate-tip-bot

The original version of the app was very simple, and i think this would be relatively simple too: https://github.com/paritytech/substrate-tip-bot/blob/5cadc69aefbd990ecf5d5f0d1a9c6f3cb38139e3/index.js

**Comment by Nadro-J** [comment]
I don't think it would be too hard to implement this :) the median value is probably the most fiddly part to do but everything else should be do-able

One of the things I've been working on in parallel is a mechanism that allows users with a specific discord role to submit feedback on a referendum and for other users with the same role to either approve or deny the request _(loosely based off multisig thresholds)_. If approved the post is made, otherwise it's denied and nothing gets posted.

The same logic could potentially be applied to a tip-bot. I don't think you can limit how many times a person can create a thread on Discord but you can certainly set limitations on commands so a user could only use /tip once every X amount of hours/days.

![image](https://github.com/user-attachments/assets/5b234f97-14b5-4331-b8f4-44e487af6cee)

![image](https://github.com/user-attachments/assets/b76a4234-65c5-4898-b59b-d2895bcd07a8)

**Comment by shawntabrizi** [comment]
Feedback form is a great idea.

Maybe over-engineering, but allowing AI to summarize the conversation inside a thread about a proposal might get similar results automatically :)

I guess the most important think is to keep anonymity, and letting the DAO control when the post happens.
